packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -94,29 +89,64 @@ class _ToolbarContainerLayout extends SingleChildLayoutDelegate {
 /// to false. In that case a null leading widget will result in the middle/title widget
 /// stretching to start.
 ///
-/// {@tool sample}
+/// {@tool snippet --template=stateless_widget_material}
+///
+/// This sample shows an [AppBar] with two simple actions. The first action
+/// opens a [SnackBar], while the second action navigates to a new page.
+///
+/// ```dart preamble
+/// final GlobalKey<ScaffoldState> scaffoldKey = GlobalKey<ScaffoldState>();
+/// final SnackBar snackBar = SnackBar(content: Text('Showing Snackbar'));
```

Comment:
Probably add a blank line below and may need `const Text()` to keep the analyzer happy.